tests/bin/check/run.t

```diff
@@ -0,0 +1,130 @@
+A necessary condition for dune-release to work is that it's able to infer the main package
+
+    $ touch first_pkg.opam
+    $ touch other_pkg.opam
+
+    $ dune-release check --working-tree
+    [FAIL] cannot determine distribution name automatically: add (name <name>) to dune-project
+    [1]
+
+Let's clean up to not interfer with the following tests
+
+    $ rm first_pkg.opam other_pkg.opam
+
+If the condition described above is fulfilled, there are 5 checks to be performed
+
+    $ cat > my_pkg.opam <<EOF \
+    > opam-version: "2.0"\
+    > dev-repo: "git+https://github.com/fu/fa.git"\
+    > description : "Some description"\
+    > EOF
+    $ echo "(lang dune 2.7)" > dune-project
+
+    $ dune-release check --dry-run --working-tree | sed "s/lint of .* and/lint of <project_dir> and/" | sed "s/.*] package(s) pass the tests/[<status>] package(s) pass the tests/" | sed '/chdir/d' | sed '/=>/d' | sed '/-: exec/d'
```

Comment:
I've tried to put these `sed` calls that get repeated several times into functions, but got an sh error. Is it possible to do that in mdx cram tests?

Reply:
I think it's simpler to define a function in a separate sh file, like I did in #333 for sanitize.sh

Reply:
That's a good idea! Also using dune cram tests instead of mdx cram tests seems great. I'll wait for that PR to be merged to use dune cram tests here then. Or is it possible to do what you're saying with mdx?

Reply:
I think it should be possible with mdx as well

Reply:
It might also be possible to mix mdx and dune cram test so we don't have to migrate as much test when doing the move. I.e. new tests should be written as dune cram tests.

Also if you need to sanitize the output, I'd be in favor of an OCaml script rather than a bash one for portablitiy of the tests and ease of maintenance!

Reply:
If we can make dune cram tests cohabitate with the current tests that would be better indeed

Reply:
Thanks to both for your advice! I've separated the functions into separate files now and have written them in OCaml.